desc/objectives/_coils.py

```diff
@@ -1282,14 +1323,25 @@ def compute(self, field_params, constants=None):
         """
         if constants is None:
             constants = self.constants
-
+        field_params = params_1 if not self._qfm_surface else params_2
```

Comment:
don't you also need to check if the field is fixed? in which case the only DOFs will be the surface?